### DISCUSSION TITLE
Customer Email Change and Order Notifications

### DISCUSSION THREAD

**Discussion by kiatng** [OpenMage/magento-lts#3289] [Ideas]
My client received complaints from customers that when they changed the emails, they no longer receive order notifications. This is because the notification takes the email address from the order's `customer_email` which was set at order creation. I can think of 2 ways to solve it:

1. Modified the method `getCustomerEmail()` in order module to take the email from the customer module. Cost is a DB read operation.
2. On customer email change, add an observer to change the `customer_email` column in DB. Cost is event-observer and a DB write.

There are thousands of notifications sent everyday. Any idea which approach is best? Is this a good feature for OM?

**Comment by m-overlund** [comment]
Sound considerations and of course customers expect to get notifications on the new email.

As far as i know, as a logged in user, you cannot make a frontend order with a custom email address.
However, in admin, you can create a customer order, with a custom email address. 

Maybe this should be considered too - what should happen here?

**Reply by kiatng** [reply]
If an order is created from backend and the user over write the email:
![image](https://github.com/OpenMage/magento-lts/assets/1106470/2f5cc1df-d5dc-4b2b-b930-cf252593112c)

and if the customer change the email later, he or she would expect to receive the notification in the new email.

**Comment by fballiano** [comment]
I'd go with option 1, it is good to have a "snapshot" of all customer data at the time of the order for bureaucratic/administrative reasons and I'd prefer the historical data never to be changed :-)

**Reply by elidrissidev** [reply]
I'd go with this one too. In addition, if customer account doesn't exist (i.e. it was deleted) `getCustomerEmail()` should default to the email address saved in the order.
